CoreFoundation/PlugIn.subproj/CFBundle_Executable.c

```diff
@@ -303,38 +275,96 @@ static CFURLRef _CFBundleCopyExecutableURLInDirectory2(CFBundleRef bundle, CFURL
     return executableURL;
 }
 
+static CFStringRef _CFStringRemovePrefixSuffix(CFStringRef str, CFStringRef prefix, CFStringRef suffix) {
+    CFMutableStringRef mutableString = CFStringCreateMutableCopy(kCFAllocatorSystemDefault, 0, str);
+    
+    // Check if the executable has shared library prefix/suffix.
+    CFRange prefixRange = CFStringFind(mutableString,
+                                       _CFBundleSharedLibraryFilenamePrefix,
+                                       kCFCompareAnchored);
+    CFRange suffixRange = CFStringFind(mutableString,
+                                       _CFBundleSharedLibraryFilenameSuffix,
+                                       kCFCompareAnchored|kCFCompareBackwards);
+    
+    // Only return the stripped string if both the prefix and suffix are found.
+    if (prefixRange.location != kCFNotFound &&
+        suffixRange.location != kCFNotFound &&
+        prefixRange.location == 0 &&
+        suffixRange.location + suffixRange.length == CFStringGetLength(mutableString)) {
+        CFStringReplace(mutableString, prefixRange, CFSTR(""));
+        
+        suffixRange = CFStringFind(mutableString,
+                                   _CFBundleSharedLibraryFilenameSuffix,
+                                   kCFCompareAnchored|kCFCompareBackwards);
+        
+        CFStringReplace(mutableString, suffixRange, CFSTR(""));
+        return mutableString;
```

Comment:
Should this function be calculating the range then just calling `CFStringCreateWithSubstring` instead?

Reply:
Yes, that's a lot better, thanks!